crates/biome_html_formatter/src/html/auxiliary/root.rs

```diff
@@ -1,20 +1,32 @@
 use crate::prelude::*;
 use biome_formatter::write;
-use biome_html_syntax::HtmlRoot;
+use biome_html_syntax::{HtmlRoot, HtmlRootFields};
 #[derive(Debug, Clone, Default)]
 pub(crate) struct FormatHtmlRoot;
 impl FormatNodeRule<HtmlRoot> for FormatHtmlRoot {
     fn fmt_fields(&self, node: &HtmlRoot, f: &mut HtmlFormatter) -> FormatResult<()> {
-        if let Some(bom) = node.bom_token() {
+        let HtmlRootFields {
+            html,
+            bom_token,
+            directive,
+            frontmatter,
+            eof_token,
+        } = node.as_fields();
+
+        if let Some(frontmatter) = frontmatter {
+            frontmatter.format().fmt(f)?;
+        }
+
+        if let Some(bom) = bom_token {
             bom.format().fmt(f)?;
         }
-        if let Some(directive) = node.directive() {
+        if let Some(directive) = directive {
             directive.format().fmt(f)?;
         }
```

Comment:
2 things here:
- the bom should be first, right?
- the frontmatter needs a hard line after it because documents end up getting formatted like `---<html>`, as seen in snapshots